ci/docker/conda-java-jni.dockerfile

```diff
@@ -0,0 +1,30 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+FROM ghcr.io/mamba-org/micromamba:ubuntu24.04
```

Comment:
I'm new to the process so this question might be dumb. Do we only build JNI libraries for ubuntu?

Reply:
No, the resulting libraries shouldn't be tied to Ubuntu. But perhaps we should use an older system as a base to avoid potential glibc incompatibility

Reply:
In any case, this is for testing and not for building the actual binaries, though

Reply:
Thanks for the explanation! Let me investigate how the actual binaries (for different platforms) are built for a release.